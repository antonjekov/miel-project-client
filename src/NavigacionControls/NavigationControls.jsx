import React from "react";
import { Navbar, Nav, Container } from 'react-bootstrap';
import logo from "./bee.png";
import styles from './NavigationControls.module.css';

function NavigationControls(props) {

    const userName = props.userName;

    return (<Navbar bg="light" expand="lg" sticky="top">
        <Container className={styles.brand}>
            <Navbar.Brand href="/">
                <img
                    src={logo}
                    width="50"
                    height="50"
                    className="d-inline-block align-top"
                    alt="Bee logo"
                />
            </Navbar.Brand>
        </Container>

        <Container className="justify-content-center">
            <Nav.Link className={styles.navLink} href="/login">Login</Nav.Link>
            <Nav.Link className={styles.navLink} href="/register">Register</Nav.Link>

            <Nav.Link className={styles.navLink} href="/logout">Logout</Nav.Link>
            <Nav.Link className={styles.navLink} href="#contacts">Contacts</Nav.Link>
            <Nav.Link className={styles.navLink} href="#articles">Articles</Nav.Link>

            <Nav.Link className={styles.navLink} href="/add-product">Add Product</Nav.Link>
            <Nav.Link className={styles.navLink} href="#contacts">Edit Product</Nav.Link>

        </Container>

        {userName?
        <Container className={styles.user}>
            <Navbar.Text >Welcome {userName}</Navbar.Text>
        </Container>:null
        }


    </Navbar>);
}

export default NavigationControls;

