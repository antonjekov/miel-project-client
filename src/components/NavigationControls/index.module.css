.navLink {
    color: black
}

.navLink:hover {
    color: rgb(9, 211, 69)
}

.containerLinks{
   align-items: center;
}

.brand{
    width: 5em;
}

.links{
    width: auto;
}

.user{
    width: 20em;
   
}





