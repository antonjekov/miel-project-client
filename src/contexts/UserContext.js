import React from "react";

const UserContext = React.createContext({name: "Anton"});

export default UserContext