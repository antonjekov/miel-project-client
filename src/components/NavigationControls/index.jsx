import React, {  Fragment } from "react";
import { Navbar, Nav, Container,Button } from 'react-bootstrap';
import logo from "./bee.png";
import styles from './index.module.css';
import { useAuth } from "../../contexts/Auth" ;
import userService from "../../services/user_service";
import { Link, useHistory } from "react-router-dom";


function NavigationControls(props) {
    const {userInfo,setUserInfo} = useAuth();
    const history = useHistory()
        
    const logout = ()=>{
        userService.logout().then(res => {
            if (!res.ok) {
                throw new Error();
            }
            setUserInfo('')
            history.push('/')
        });
    }

    const shoppingCardClickHandler = ()=>{
        history.push('/shoppingCard')
    }

    return (

        <Navbar bg="light" expand="lg" >
            <Container className={styles.brand}>
                <Navbar.Brand as={Link} to="/">
                    <img
                        src={logo}
                        width="50"
                        height="50"
                        className="d-inline-block align-top"
                        alt="Bee logo"
                    />
                </Navbar.Brand>
            </Container>

            <Container className="justify-content-center">
                {userInfo?
                <Nav.Link  className={styles.navLink} onClick={logout}>Logout</Nav.Link>:

                <Fragment>
                <Nav.Link as ={Link} className={styles.navLink} to="/login">Login</Nav.Link>
                <Nav.Link as ={Link} className={styles.navLink} to="/register">Register</Nav.Link> 
                </Fragment>                 
                
                }

                <Nav.Link as ={Link} className={styles.navLink} to="/contacts">Contacts</Nav.Link>

                {userInfo&&userInfo.role==='admin'?
                <Fragment>
                <Nav.Link as ={Link} className={styles.navLink} to="/add-product">Add Product</Nav.Link>
                <Nav.Link as ={Link} className={styles.navLink} to="/add-subcategory">Add Subcategory</Nav.Link>
                </Fragment>:''
                }

            </Container>
            {userInfo?(<Container className={styles.user}>
                <Navbar.Text >Welcome {userInfo.name}</Navbar.Text>
            </Container>):''}
            
            {userInfo&&userInfo.role==='client'?<Button variant="warning" onClick={shoppingCardClickHandler} ><b>Shopping cart</b> products <b>{userInfo.shoppingCard.length}</b></Button>:''}
            
        </Navbar>
    );
}

export default NavigationControls;

