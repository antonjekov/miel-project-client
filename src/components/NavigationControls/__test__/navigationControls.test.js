import React from "react";
import ReactDOM from "react-dom"
import {render, cleanup} from "@testing-library/react"
import renderer from "react-test-renderer"
import NavigationControls from "../"
import { useAuth, AuthContext } from "../../../contexts/Auth";

const userAdmin = {"_id":"5f073d023af80439e4f51c2a","name":"admin","username":"admin@yahoo.com","role":"admin","shoppingCard":[]}

const userClient = {"_id":"5f073d613af80439e4f51c2b","name":"anton","username":"anton@yahoo.com","role":"client","shoppingCard":["5f073f0c3af80439e4f51c2c","5f073f0c3af80439e4f51c2c"]}

afterEach(cleanup)

it('renders without crashing', ()=>{
    const div = document.createElement('div');
    ReactDOM.render(<AuthContext.Provider value={{ userInfo:userAdmin}}>
                            <NavigationControls />
                        </AuthContext.Provider>                    
                    ,div)
})

//Snapshot tests
it('matches snapshot admin', ()=>{
    const tree = renderer.create(
        <AuthContext.Provider value={{ userInfo:userAdmin}}>
            <NavigationControls/>
        </AuthContext.Provider>
   ).toJSON();
    expect(tree).toMatchSnapshot();
})

it('matches snapshot client', ()=>{
    const tree = renderer.create(
        <AuthContext.Provider value={{ userInfo:userClient}}>
            <NavigationControls/>
        </AuthContext.Provider>
   ).toJSON();
    expect(tree).toMatchSnapshot();
})
it('matches snapshot without user', ()=>{
    const tree = renderer.create(
        <AuthContext.Provider value={{ userInfo:null}}>
            <NavigationControls/>
        </AuthContext.Provider>
   ).toJSON();
    expect(tree).toMatchSnapshot();
})