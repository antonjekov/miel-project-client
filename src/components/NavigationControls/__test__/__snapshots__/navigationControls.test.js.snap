// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot admin 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <div
    className="brand container"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      <img
        alt="Bee logo"
        className="d-inline-block align-top"
        height="50"
        src="bee.png"
        width="50"
      />
    </a>
  </div>
  <div
    className="justify-content-center container"
  >
    <a
      className="navLink nav-link"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      Logout
    </a>
    <a
      className="navLink nav-link"
      href="/contacts"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      Contacts
    </a>
    <a
      className="navLink nav-link"
      href="/add-product"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      Add Product
    </a>
    <a
      className="navLink nav-link"
      href="/add-subcategory"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      Add Subcategory
    </a>
  </div>
  <div
    className="user container"
  >
    <span
      className="navbar-text"
    >
      Welcome 
      admin
    </span>
  </div>
  
</nav>
`;

exports[`matches snapshot client 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <div
    className="brand container"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      <img
        alt="Bee logo"
        className="d-inline-block align-top"
        height="50"
        src="bee.png"
        width="50"
      />
    </a>
  </div>
  <div
    className="justify-content-center container"
  >
    <a
      className="navLink nav-link"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      Logout
    </a>
    <a
      className="navLink nav-link"
      href="/contacts"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      Contacts
    </a>
    
  </div>
  <div
    className="user container"
  >
    <span
      className="navbar-text"
    >
      Welcome 
      anton
    </span>
  </div>
  <a
    className="btn btn-warning"
    href="/shoppingCard"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <b>
      Shopping cart
    </b>
     products 
    <b>
      2
    </b>
  </a>
</nav>
`;

exports[`matches snapshot without user 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <div
    className="brand container"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      <img
        alt="Bee logo"
        className="d-inline-block align-top"
        height="50"
        src="bee.png"
        width="50"
      />
    </a>
  </div>
  <div
    className="justify-content-center container"
  >
    <a
      className="navLink nav-link"
      href="/login"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      Login
    </a>
    <a
      className="navLink nav-link"
      href="/register"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      Register
    </a>
    <a
      className="navLink nav-link"
      href="/contacts"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      Contacts
    </a>
    
  </div>
  
  
</nav>
`;
